vllm_spyre/compilation_utils.py

```diff
@@ -0,0 +1,228 @@
+            f"pre-compiled model config was found in the "
+            f"TORCH_SENDNN_CACHE_DIR: {str(compilation_config_path)} or"
+            f"{str(compilation_catalog_path)} does not exist")
+
+    if not compilation_catalog_path.is_file() and \
+        not compilation_config_path.is_file():
+        raise ValueError(
+            "{req_precompiled_decoder_env_var}=1 was set, but the "
+            "pre-compiled model config is not a file")
+
+    matching_config = None
+
+    # Note: In below implementation we don't tell user exactly what's wrong
+    # but we do "warn" them about mismatch and provide the list of supported
+    # configuration along with what they have given us.
+    if compilation_catalog_path.is_file():
+        with open(compilation_catalog_path) as f:
+            try:
+                pre_compile_catalog = json.load(f)
+            except json.JSONDecodeError as e:
+                raise ValueError(
+                    f"Precompiled catalog {str(compilation_catalog_path)}"
+                    " is not a valid JSON file") from e
+        match_result = match_from_pre_compile_catalog(pre_compile_catalog,
+                                                      vllm_config)
+
+        if match_result == -1:
+            # No match found
+            logger.warning(
+                "[PRECOMPILED_WARN]-" \
+                "Provided vllm configuration doesn't match any of the "
+                "pre-compiled model configurations. Catalog: \n%s\n "
+                "vllm_config: \n%s", str(compilation_catalog_path),
+                str(vllm_config))
+
+            # Return with warning
+            return
+        else:
+            matching_config = pre_compile_catalog[match_result]
+
+    elif compilation_config_path.is_file():
+        with open(compilation_config_path) as f:
+            try:
+                compilation_config = json.load(f)
+            except json.JSONDecodeError as e:
+                raise ValueError("Precompiled model config "
+                                 f"{str(compilation_config_path)} was "
+                                 "not valid json") from e
+        match_result = match_from_model_config_file(compilation_config,
+                                                    vllm_config)
+        if not match_result:
+            logger.warning(
+                "[PRECOMPILED_WARN]-"
+                "Provided vllm configuration doesn't match any of the "
+                "pre-compiled model")
+            # Return with warning
+            return
+        else:
+            matching_config = compilation_config
+
+    if matching_config:
+        # Check vllm_spyre version
+        try:
+            from vllm_spyre._version import version as vllm_spyre_version
+            if matching_config["vllm_spyre_version"] != vllm_spyre_version:
+                # Can be converted to ValueError if we want to be strict
+                # with checking
```

Comment:
Now that the configuration is expcility `VLLM_SPYRE_REQUIRE_PRECOMPILED_DECODERS` we could be stricter on the checking here for cases where we know that no entry in the catalog can work (version mistmatches).

Reply:
NVM, the `vllm_spyre_version` doesn't necessarily invalidate the cache...